test/e2e/fqdn_dns_cache_test.go

```diff
@@ -0,0 +1,314 @@
+// Copyright 2024 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"bytes"
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+	"text/template"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+
+	crdv1beta1 "antrea.io/antrea/pkg/apis/crd/v1beta1"
+	agentconfig "antrea.io/antrea/pkg/config/agent"
+	"antrea.io/antrea/test/e2e/utils"
+)
+
+func TestFQDNPolicyWithCachedDNS(t *testing.T) {
+	const (
```

Comment:
Having constants in a function like this seem very weird to me. I think you can just make these variables instead

Reply:
why make them variables if they are meant to be constant values? Declaring them in the function just limits the scope and avoids name conflicts as we have a lot of top-level constants / variables in the e2e files. Keeping the scope narrow is good IMO, and I haven't encountered guidance around declaring such constants before.

Reply:
Sure, then maybe in other e2e files we can check if we could follow the same style and eliminate some top-level constants that are only used for certain functions

Reply:
I would think we already have a few other examples in the codebase of such usage. It's probably not worth reviewing all existing test code to update it, but something to keep in mind for future tests.